src/core/Song.cpp

```diff
@@ -775,10 +775,11 @@ void Song::stopExport()
 void Song::insertBar()
 {
 	m_tracksMutex.lockForRead();
-	for( TrackList::const_iterator it = tracks().begin();
-					it != tracks().end(); ++it )
+	for (Track* track: tracks())
 	{
-		( *it )->insertBar( m_playPos[Mode_PlaySong] );
+		// FIXME journal batch of tracks instead of each track individually
+		track->addJournalCheckPoint();
```

Comment:
```suggestion
		if (track->numOfClips() > 0){ track->addJournalCheckPoint(); }
```
And same for removal

Reply:
Good idea. Added.

Still this is no guarantee that all undo will have a visible effect here. Instead of checking for that we should implement batching of checkpoints...